dns_config/test/api_acl_test.go

```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
 
+		readRequest := aclAPI.AclAPI.AclRead(ctx, *dummyAcl.Id)
+		resp, httpRes, err := aclAPI.AclAPI.AclReadExecute(readRequest)
 		require.Nil(t, err)
 		require.NotNil(t, resp)
-		assert.Equal(t, 200, httpRes.StatusCode)
+		require.Equal(t, 200, httpRes.StatusCode)
+		require.Equal(t, dummyAcl, *resp.Result)
+	})
+
+	t.Run("Test AclAPIService AclList", func(t *testing.T) {
 
+		//t.Skip("skip test") // remove to run test
+
+		dummyAclList := []dns_config.ConfigACL{
+			{
+				Comment: openapiclient.PtrString("This is a dummy ACL for testing."),
+				Id:      openapiclient.PtrString("dummyAclId1"),
+				Name:    "dummyAclName1",
+			},
+			{
+				Comment: openapiclient.PtrString("This is another dummy ACL for testing."),
+				Id:      openapiclient.PtrString("dummyAclId2"),
+				Name:    "dummyAclName2",
+			},
+		}
+
+		testClient := NewTestClient(func(req *http.Request) *http.Response {
+			require.Equal(t, "GET", req.Method)
+			require.Equal(t, "/api/ddi/v1/dns/acl", req.URL.Path)
+
+			response := dns_config.ConfigListACLResponse{
+				Results: dummyAclList,
+			}
+			body, err := json.Marshal(response)
+			require.NoError(t, err)
+
+			return &http.Response{
+				StatusCode: 200,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+				Header:     map[string][]string{"Content-Type": {"application/json"}},
+			}
+		})
+
+		configuration := internal.NewConfiguration()
+		configuration.HTTPClient = testClient
+
+		aclAPI := dns_config.NewAPIClient(configuration)
+		ctx := context.Background()
+
+		resp, httpRes, err := aclAPI.AclAPI.AclList(ctx).Execute()
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, 200, httpRes.StatusCode)
+		require.Equal(t, dummyAclList, resp.Results)
 	})
 
 	t.Run("Test AclAPIService AclUpdate", func(t *testing.T) {
 
-		t.Skip("skip test") // remove to run test
+		//t.Skip("skip test") // remove to run test
+
+		dummyAcl := dns_config.ConfigACL{
+			Comment: openapiclient.PtrString("This is a dummy ACL for testing."),
+			Id:      openapiclient.PtrString("dummyAclId"),
+			Name:    "dummyAclName",
+		}
+
+		updatedAcl := dns_config.ConfigACL{
+			Comment: openapiclient.PtrString("This is an updated dummy ACL for testing."),
+			Id:      dummyAcl.Id,
+			Name:    "updatedDummyAclName",
+		}
+
+		testClient := NewTestClient(func(req *http.Request) *http.Response {
+			require.Equal(t, "PATCH", req.Method)
+			require.Equal(t, "/api/ddi/v1/dns/acl/"+*dummyAcl.Id, req.URL.Path)
+
+			response := dns_config.ConfigUpdateACLResponse{
+				Result: &updatedAcl,
+			}
+			body, err := json.Marshal(response)
+			require.NoError(t, err)
+
+			return &http.Response{
+				StatusCode: 200,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+				Header:     map[string][]string{"Content-Type": {"application/json"}},
+			}
+		})
+
+		configuration := internal.NewConfiguration()
+		configuration.HTTPClient = testClient
+
+		aclAPI := dns_config.NewAPIClient(configuration)
+		ctx := context.Background()
```

Comment:
you could have a common function for this in each package. 

```
func newAPITestClient(testFn RoundTripFunc) *dns_config.APIClient {
    configuration := internal.NewConfiguration()
    configuration.HTTPClient = NewTestClient(testFn)

   return dns_config.NewAPIClient(configuration)
}
```
That way you can avoid some duplication and simply write 

```
dnsConfigAPI := newAPITestClient(func(req *http.Request) *http.Response {
	require.Equal(t, "PATCH", req.Method)
        .....
}